libtelemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryService.java

```diff
@@ -48,8 +47,10 @@ public int onStartCommand(Intent intent, int flags, int startId) {
 
   @Override
   public void onDestroy() {
-    // TODO If the location receiver was not created (permissions were not granted) do not unregister
-    unregisterLocationReceiver();
+    if (locationReceiver != null) {
```

Comment:
I think this is not necessary because if we never create `TelemetryService` we can't get a `null` `locationReceiver`. This is because how we're dealing with the `TelemetryService` creation in `MapboxTelemetry`. Am I missing something?